finality-aleph/src/sync/mod.rs

```diff
@@ -64,6 +64,8 @@ pub trait Justification: Clone + Send + Sync + Debug + 'static {
     fn into_unverified(self) -> Self::Unverified;
 }
 
+type BlockIdForBlock<B> = <<B as Block>::Header as Header>::Identifier;
```

Comment:
Is this really necessary? I would expect all places that have `Block` to also have `Justification`, so you can just use the other one.

If it is, then at least restore some symmetry making the other one `BlockIdForJustification` or change the name somehow. Maybe this could be `IdFor`, since the block is the argument?